src/components/AnimatedSubmitButton/index.tsx

```diff
@@ -0,0 +1,137 @@
+import React, {useCallback, useEffect, useMemo, useRef, useState} from 'react';
+import Animated, {Keyframe, runOnJS, useAnimatedStyle, useSharedValue, withTiming} from 'react-native-reanimated';
+import Button from '@components/Button';
+import * as Expensicons from '@components/Icon/Expensicons';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import variables from '@styles/variables';
+import CONST from '@src/CONST';
+
+type AnimatedSubmitButtonProps = {
+    // Whether to show the success state
+    success: boolean | undefined;
+
+    // Text to show on the button
+    text: string;
+
+    // Function to call when the button is pressed
+    onPress: () => void;
+
+    // Whether the animation is running
+    isSubmittingAnimationRunning: boolean;
+
+    // Function to call when the animation finishes
+    onAnimationFinish: () => void;
+};
+
+function AnimatedSubmitButton({success, text, onPress, isSubmittingAnimationRunning, onAnimationFinish}: AnimatedSubmitButtonProps) {
```

Comment:
Any reason we have this button be specific to the submit action instead of a generic animated button that can take the durations and text as prop so that I can be reused elsewhere in App?